### PR TITLE
Overhauled UserTrialPermissions

### PR DESCRIPTION
**Pull request by marshl**
Removed the Primary Investigator role and replaced it with the Create Trial and View Trial roles
Changed trial permissions to create a new permission for the owner of the trial when the trial is created
Changed the trial permissions screen to prevent users that don't have the View Trial role from being added
Removed the unused Trial.search() and TrialPatient.search) methods

### CODE REVIEW

File: models/Trial.php [hunk 1]
```diff
@@ -267,36 +257,31 @@ public static function checkTrialAccess($user, $trial_id, $permission)
...
+    /**
+     * @param CWebUser $user The user to get access for
+     * @return int The user permission if they have one otherwise null)
+     */
+    public function getTrialAccess($user)
     {
-        if ($this->owner_user_id === $user_id) {
-            return UserTrialPermission::PERMISSION_MANAGE;
+        if (!$user->checkAccess('TaskViewTrial')) {
+            echo 'erhmerghesh';
```
Comment (Vegemash) [inline models/Trial.php]: perhaps this could be more descriptive?

File: controllers/TrialController.php [hunk 2]
```diff
@@ -392,4 +457,54 @@ public function actionGetTrialList($trialID, $type)
 
         echo '<div class="large-3 column trial-list">' . $dropDown . '</div>';
```
Comment (apigram) [inline controllers/TrialController.php]: Though this is correct, I'm wondering if the echo is actually necessary here or whether just putting the HTML in there will still have the same effect.

File: models/Trial.php [hunk 3]
```diff
@@ -256,8 +223,31 @@ public static function model($className = __CLASS__)
...
+            }
+        }
+    }
+
+    /**
      * Returns whether or not the given user can access the given trial using the given action
-     * @param $user User The user to check access for
+     * @param $user CWebUser The user to check access for
      * @param $trial_id int The ID of the trial
      * @param $permission integer The ID of the controller action
```
Comment (apigram) [inline models/Trial.php]: Might be worth keeping the PHPDoc annotations consistent. Should be one of either int or integer. You decide.

File: models/Trial.php (hunk 1 above)
Comment (marshl) [inline models/Trial.php]: whoops

File: views/trial/permissions.php [hunk 4]
```diff
@@ -141,26 +152,22 @@ function removeSelectedUser() {
         'type': 'GET',
         'url': '<?php echo $this->createUrl('addPermission'); ?>',
         'data': {id: <?php echo $model->id; ?>, user_id: $('#user_id').val(), permission: $('#permission').val()},
-          'success'
-      :
-      function (html) {
-        if (html == <?php echo TrialController::RETURN_CODE_USER_PERMISSION_ALREADY_EXISTS; ?>) {
+        'success': function (html) {
+          if (html == <?php echo TrialController::RETURN_CODE_USER_PERMISSION_ALREADY_EXISTS; ?>) {
+            new OpenEyes.UI.Dialog.Alert({
```
Comment (apigram) [inline views/trial/permissions.php]: I really should start using these for the AJAX functions I define as well.

File: controllers/TrialController.php [hunk 5]
```diff
@@ -392,4 +457,54 @@ public function actionGetTrialList($trialID, $type)
...
     }
+
+    /**
+     * Quries users that can be assigned to the Trial and that match the search term.
```
Comment (marshl) [inline controllers/TrialController.php]: Typo "Quries" => "Queries"

File: views/trial/permissions.php (hunk 4 above)
Comment (marshl) [inline views/trial/permissions.php]: It does feel a little hacky though, maybe I should actually return a JSON encoded array with an error number and a message.

File: models/Trial.php (hunk 3 above)
Comment (marshl) [inline models/Trial.php]: Good idea, I'll also replace any uses of "$var type" with "type $var" to keep consistent

File: views/trial/_trialList.php [hunk 6]
```diff
@@ -30,12 +30,17 @@
...
         <thead>
         <tr>
-            <?php foreach (array('Name', 'Date Created', 'Owner') as $i => $field) { ?>
+            <?php foreach (array('Name', 'Date Created', 'Owner', 'Status') as $i => $field) { ?>
               <th id="patient-grid_c<?php echo $i; ?>">
                   <?php
                   $new_sort_dir = ($i == $sort_by) ? 1 - $sort_dir : 0;
+                  $sort_symbol = '';
+                  if ($i == $sort_by) {
+                      $sort_symbol = $sort_dir == 1 ? '&#x25B2;' : '&#x25BC';
```
Comment (Vegemash) [inline views/trial/_trialList.php]: Can we not have magic numbers without explanation here please?

Comment (marshl) [inline views/trial/_trialList.php]: Would you prefer the ↑ (&uarr;), and ↓ (&darr;) arrows over the ▲▼ arrows ? Or should I just add a comment in the code?

**Pull request merged**